static/panes/compiler.js

```diff
@@ -2284,21 +2284,47 @@ var hexLike = /^(#?[$]|0x)([0-9a-fA-F]+)$/;
 var hexLike2 = /^(#?)([0-9a-fA-F]+)H$/;
 var decimalLike = /^(#?)(-?[0-9]+)$/;
 
+function parseNumericValue(value) {
+    var hexMatch = hexLike.exec(value) || hexLike2.exec(value);
+    if (hexMatch)
+        return bigInt(hexMatch[2], 16);
+
+    var decMatch = decimalLike.exec(value);
+    if (decMatch)
+        return bigInt(decMatch[2]);
+
+    return null;
+}
+
 function getNumericToolTip(value) {
-    var match = hexLike.exec(value) || hexLike2.exec(value);
-    if (match) {
-        return value + ' = ' + bigInt(match[2], 16).toString(10);
-    }
-    match = decimalLike.exec(value);
-    if (match) {
-        var asBig = bigInt(match[2]);
-        if (asBig.isNegative()) {
-            asBig = bigInt('ffffffffffffffff', 16).and(asBig);
+    var numericValue = parseNumericValue(value);
+    if (numericValue === null)
+        return null;
+
+    // Decimal representation.
+    var result = numericValue.toString(10);
+
+    // Hexadecimal representation.
+    if (numericValue.isNegative()) {
+        var masked = bigInt('ffffffffffffffff', 16).and(numericValue);
+        result += ' = 0x' + masked.toString(16).toUpperCase();
+    } else {
+        result += ' = 0x' + numericValue.toString(16).toUpperCase();
+    }
+
+    // ASCII character.
+    if (numericValue.greaterOrEquals(0) && numericValue.lesserOrEquals(0x7F)) {
+        var char = String.fromCharCode(numericValue.valueOf());
+        if (('A' <= char && char <= 'Z')
+            || ('a' <= char && char <= 'z')
+            || ('0' <= char && char <= '9')
+            || "!\"#$%&'()*+,-./:;<=>?@[] ^_`{|}~ ".includes(char)) {
```

Comment:
I haven't found a better way to check whether a character is printable than the big if condition, but my Google skills might have failed me.

Reply:
how about comparing the decimal values instead? so basically (numericValue >= 32 && numericValue <= 126) https://commons.wikimedia.org/wiki/File:ASCII-Table-wide.svg

Reply:
... I have written three iterations of ASCII matchers for lexy in the past year in exactly the same way, but forgot that I could do it here. :D